VP-1310: Adapt AuthorizeNet for platform 3.0.0

yecli opened the pull request:
Decided to not create .Core project for the single ModuleConstants and .Data project for the single AuthorizeNetMethod implementation class. 
If better to separate them - please tell.
Also thinking on extracting service for huge API controller `RegisterPayment` method.

OlegoO committed:
[draft] Updated License

yecli committed:
Merge remote-tracking branch 'origin/dev'

kostyrin committed:
#VP-1157 Response saved to Transactions

kostyrin committed:
#vp-1157 csproj

akak1977 committed:
Codestyle changes. webClient came disposed after use.

akak1977 committed:
Merge branch 'feature/VP-1157-auth-net-transaction-info'

akak1977 committed:
2.13.2

yecli committed:
VP-1310: Migrate to V3

- Structure changed;
- Adopted code;
- Settings addition in process;

yecli committed:
VP-1310: Remove artifacts folder

yecli committed:
VP-1310: Add artifacts to .gitignore

yecli committed:
VP-1310: Actualize AuthorizeNet for v3

yecli committed:
VP-1310: Fix "thank you" page redirect

- Changed ThankYouPageRelativeUrl to ThankYouPageAbsoluteUrl as we cannot get storeUrl now;

yecli committed:
VP-1310: Merging with release/3.0.0

yecli committed:
VP-1310: Remove redundat solution file

yecli committed:
VP-1310: Fix ThankYouPageUrl default value template

yecli committed:
VP-1310: Change Jenkinsfile to V3

yecli committed:
VP-1310: Fix possible NullReference

yecli committed:
VP-1310: Simplify method registration

Factory is not needed as the is no NullReference exeptions with settings now;

yecli committed:
VP-1310: Support absolute URL for ThankYouPageUrl

- User can specify absolute ThankYouPageUrl, while we remove scheme in code to pass to AuthorizeNet for redirection. If it passed with the scheme, redirect is not working.

yecli committed:
VP-1310: Move secure settings to configuration

- To avoid passing secure sensitive data through API, we should not store secure settings in platform settings. They should be set in configuration in specific section. For AuthorizeNET payment method add the following values to appsettings.json:
```
 "Payments": {
  "AuthorizeNet": {
   "ApiLogin": "***",
   "TxnKey": "***",
   "SHA2Hash": "***"
  }
 }
```

akak1977 commented on src/VirtoCommerce.AuthorizeNet.Web/Controllers/Api/AuthorizeNetController.cs:
```diff
@@ -0,0 +1,101 @@
+        [HttpPost]
+        [Route("registerpayment/{orderId}")]
+        [AllowAnonymous]
+        public async Task<ActionResult> RegisterPayment(string orderId)
```
Please mark action with swagger-related attributes.
```csharp
        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
```

yecli committed:
VP-1310: Add ProducesResponseType

yecli committed:
VP-1310: Move logic from AuthorizeNetController

- Refactored `AuthorizeNetController` to use `IAuthorizeNetRegisterPaymentService` logic;

yecli committed:
VP-1310: Update readme

- Described changes in options

yecli committed:
VP-1310: Change readme

- Corrected sentence;
- Fixed typos;

tatarincev commented on src/VirtoCommerce.AuthorizeNet.Web/module.manifest:
```diff
@@ -0,0 +1,30 @@
+    <platformVersion>3.0.0</platformVersion>
+    <dependencies>
+      <dependency id="VirtoCommerce.Store" version="3.0.0" />
+			<dependency id="VirtoCommerce.Orders" version="3.0.0-rc.3-001" />
```
Use version="3.0.0" for all dependencies

yecli replied:
Done

tatarincev commented on src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs:
```diff
@@ -0,0 +1,84 @@
+
+namespace VirtoCommerce.AuthorizeNet.Web.Services
+{
+    public class AuthorizeNetRegisterPaymentService : IAuthorizeNetRegisterPaymentService
```
Why do we need this separated abstraction, is it impossible to move into PaymentMethod?

yecli replied:
It defines payment result handling workflow. 
It was extracted from the old controller to facilitate controller-contained logic:
https://github.com/VirtoCommerce/vc-module-Authorize.Net/pull/5/files/6f5afbab40be3ff01eff211e68743dc1d900e594#diff-9a6d5d29037c4d28b6805d97050723afL33

Not sure we need to move it to PaymentMethod, as it has the methods for specific actions (Process, PostProcess, Refund etc.). New abstraction is like a manager calling different PaymentMethod's operations.
Almost the same code excluding AuthorizeNet specific and return type exists in Orders module:
https://github.com/VirtoCommerce/vc-module-order/blob/release/3.0.0/src/VirtoCommerce.OrdersModule.Web/Controllers/Api/OrderModuleController.cs#L419

So I think we could have some abstraction for this logic to use it in all places.

tatarincev replied:
Ok, then. Anyway, we will change this logic soon into an absolutely different manner.

yecli committed:
VP-1310: Change dependencies to 3.0.0
